### PR TITLE
DEVPROD-10020: Open failing line section alongside section containing the share line

### PR DESCRIPTION
**Pull request by SupaJoon**
DEVPROD-10020

### Description
These code changes are based off of https://github.com/evergreen-ci/ui/pull/328. Currently, the share line section will open when both a failing line and a bookmark exist. These code changes allow both sections to open when both exist.

### CODE REVIEW

File: apps/parsley/src/context/LogContext/index.tsx [hunk 1]
```diff
@@ -155,7 +155,9 @@ const LogContextProvider: React.FC<LogContextProviderProps> = ({
   const sectioning = useSections({
     logType: state.logMetadata?.logType,
     logs: state.logs,
-    onInitOpenSectionContainingLine: shareLine ?? state.failingLine,
+    onInitOpenSectionsContainingLines: [shareLine, state.failingLine].filter(
+      (v): v is number => v !== undefined,
```
Comment (sophstad) [inline apps/parsley/src/context/LogContext/index.tsx:159]: Wouldn't `v !== undefined` return a boolean? 
```suggestion
      (v): boolean => v !== undefined,
```

File: apps/parsley/src/hooks/useSections/useSections.test.tsx [hunk 2]
```diff
@@ -305,13 +305,13 @@ describe("useSections", () => {
     });
   });
 
-  it("should open the section containing 'onInitOpenSectionContainingLine' during initialization only", async () => {
+  it("should open the section containing 'onInitOpenSectionsContainingLines' during initialization only", async () => {
```
Comment (sophstad) [inline apps/parsley/src/hooks/useSections/useSections.test.tsx:308]: Should a test be added with an `onInitOpenSectionsContainingLines` array of > 1 element?

**Review by sophstad — APPROVED** [review]

File: apps/parsley/src/context/LogContext/index.tsx (hunk 1 above)
Comment (SupaJoon) [reply apps/parsley/src/context/LogContext/index.tsx:159]: the filter function still returns a boolean. `v is number` is a type predicate and it prevents me from having to cast the result of the filter to a number array.

File: apps/parsley/src/hooks/useSections/useSections.test.tsx (hunk 2 above)
Comment (SupaJoon) [reply apps/parsley/src/hooks/useSections/useSections.test.tsx:308]: I updated at test to include an array with length > 1. I test this further in populateSectionState tests [here](https://github.com/evergreen-ci/ui/pull/333/files#diff-a2ce52fa21af9fc02b6ed9726c865b4e6a7694d4c2fb65309a77af3feb73407bR572)

**Pull request merged**